libethcore/EthashGPUMiner.cpp

```diff
@@ -85,14 +91,33 @@ class EthashCLHook: public ethash_cl_miner::search_hook
 //		std::cerr << "Searched " << _count << " from " << _startNonce << std::endl;
 		m_owner->accumulateHashes(_count);
 		m_last = _startNonce + _count;
-		if (m_abort || m_owner->shouldStop())
-			return (m_aborted = true);
-		return false;
+		//if (m_abort || m_owner->shouldStop())
+		//	return (m_aborted = true);
+		//return false;
+		// TODO: still need a way to stop the GPU thread
```

Comment:
To simplify: the main kernel is `while(true) { if (abort); break; ...}`, the second one is just `abort = true;`.

But I don't get why you need the second kernel to set to abort flag. A buffer shared with CPU is not enough?